File: vipimo/matriki/bench.c
#include <stdio.h>
#include <stdlib.h>

#define N 300

int main() {
    int* a = malloc(sizeof(int) * N * N);
    int* b = malloc(sizeof(int) * N * N);
    int* c = malloc(sizeof(int) * N * N);
    for (int i = 0; i < N * N; i++) {
        a[i] = (i % 97) + 1;
        b[i] = (i % 89) + 1;
        c[i] = 0;
    }
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            int jumla = 0;
            for (int k = 0; k < N; k++) {
                jumla += a[i * N + k] * b[k * N + j];
            }
            c[i * N + j] = jumla;
        }
    }
    long long hesabu = 0;
    for (int i = 0; i < N * N; i++) hesabu += c[i];
    printf("hesabu=%lld c00=%d\n", hesabu, c[0]);
    return 0;
}

File: vipimo/matriki/bench.rs
const N: usize = 300;

fn main() {
    let mut a = vec![0i32; N * N];
    let mut b = vec![0i32; N * N];
    let mut c = vec![0i32; N * N];
    for i in 0..N * N {
        a[i] = (i % 97) as i32 + 1;
        b[i] = (i % 89) as i32 + 1;
    }
    for i in 0..N {
        for j in 0..N {
            let mut jumla: i32 = 0;
            for k in 0..N {
                jumla += a[i * N + k] * b[k * N + j];
            }
            c[i * N + j] = jumla;
        }
    }
    let mut hesabu: i64 = 0;
    for i in 0..N * N { hesabu += c[i] as i64; }
    println!("hesabu={} c00={}", hesabu, c[0]);
}
